### ISSUE TITLE
Git bash is missing icon on taskbar.

### ISSUE THREAD

**Issue by khanhoct** [git-for-windows/git#819]
### Setup

 - Which version of Git for Windows are you using? 32-bit or 64-bit? Include the
   output of `git version` as well.

Git 2.9.0 - 64bit

```
$ git --version
git version 2.9.0.windows.1
```

 - Which version of Windows are you running? 32-bit or 64-bit?

Windows 10 Pro - 64 bit

 - What options did you set as part of the installation? Or did you choose the
   defaults?

Defaults

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Bash

 - What is the issue?
Git bash is missing icon on taskbar. I've been tried open Git CMD, it is normally. This issues is adversely affect user experience.

![image](https://cloud.githubusercontent.com/assets/13175175/16828073/b753a2ba-49b6-11e6-9f3e-915e7ff8178a.png)

**Comment by dscho** [comment]
This is probably an issue with MinTTY.

We have a track record of problems with this component, and one reason I did not yet upgrade to the latest version of that component is that I dread the prospect: I fully expect having to spend tons of time to figure out how to solve new problems.

So if you have time, it would be really good if you could give me a hand. Do you have time?

**Comment by khanhoct** [comment]
Yes. I have time but I don't know how to help you.

**Comment by dscho** [comment]
@khanhoct thank you for your willingness to help.

These steps should get you started:

1. install the [Git for Windows SDK](https://git-for-windows.github.io/#download-sdk),
2. Download [MSYS2's MinTTY package](http://repo.msys2.org/msys/x86_64/mintty-2.0.1-2-x86_64.pkg.tar.xz) (it is shadowed by Git for Windows' custom version),
3. Install that package into your SDK by calling `pacman -U /path/to/mintty-2.0.1-2-x86_64.pkg.tar.xz` (if you closed the SDK window, you can reopen it via the shortcut that was added onto your Desktop),
4. run `git-bash.exe` from your SDK's top-level directory and determine whether this works and fixes the task bar issue.

Thank you!

**Comment by khanhoct** [comment]
I fail at the fist step. When I run "git-sdk-installer-1.0.3-64.7z.exe", a CMD windows show that:

```
:: Synchronizing package databases...
error: failed retrieving file 'git-for-windows.db' from dl.bintray.com : Failed to connect to dl.bintray.com port 443: Connection timed out
error: failed to update git-for-windows (download library error)
 mingw32                                        281.2 KiB   231K/s 00:01 [#######################################] 100%
 mingw32.sig                                     96.0   B  93.8K/s 00:00 [#######################################] 100%
 mingw64                                        280.7 KiB   116K/s 00:02 [#######################################] 100%
 mingw64.sig                                     96.0   B  93.8K/s 00:00 [#######################################] 100%
 msys                                           135.6 KiB   114K/s 00:01 [#######################################] 100%
 msys.sig                                        96.0   B  93.8K/s 00:00 [#######################################] 100%
error: failed to synchronize all databases

There was a problem accessing the MSYS2 repositories
If your setup requires an HTTP proxy to access the web,
please specify it here, otherwise leave it empty.

HTTP proxy:
```
A cursor flash after HTTP proxy. Maybe it asked me to enter proxy.

**Comment by dscho** [comment]
Sorry to hear that you have problems. It looks like BinTray times out. Maybe it will succeed at a later date?

**Comment by khanhoct** [comment]
@dscho : Today, I installed git-sdk-installer-1.0.3-64.7z.exe successfully. At step 3, I see the following:

```
khanh@khanh-windows MINGW64 /
$ pacman -U C:/Users/khanh/Downloads/mintty-2.0.1-2-x86_64.pkg.tar.xz
loading packages...
warning: downgrading package mintty (1~2.0.3-2 => 2.0.1-2)
resolving dependencies...
looking for conflicting packages...
error: failed to prepare transaction (could not satisfy dependencies)
:: msys2-launcher-git: installing mintty (2.0.1-2) breaks dependency 'mintty>=2.2.1'
```

**Comment by dscho** [comment]
> 'mintty>=2.2.1'

That probably means that there is a newer package available from MSYS2. I was under the impression that new packages/package versions would be uploaded exclusively to http://repo.msys2.org/msys/x86_64/, but apparently I was wrong: the newest MinTTY package seems to be hosted on SourceForge: https://sourceforge.net/projects/msys2/files/REPOS/MSYS2/x86_64/mintty-1%7E2.4.0-1-x86_64.pkg.tar.xz/download

Sorry for that!

**Comment by khanhoct** [comment]
@dscho : I completed. But it seem the problem still exists.

**I discovered that if you run git-bash for the fist time, this problem doesn't occur. It happen when run the next time.** 

![capture](https://cloud.githubusercontent.com/assets/13175175/17078870/3945b91a-5129-11e6-97b2-24714d8ac03d.PNG)

**Comment by dscho** [comment]
@khanhoct thank you for trying. Seems that my concerns with MinTTY were merited.

Last time I had to look into MinTTY issues, it took me dozens of hours to get to the bottom of the issue, and my PR with the fix was first ignored for a couple of months, then it took ages to convince MinTTY's maintainer that there *was* an issue to begin with, and then an alternative patch (with an incompatible solution) was applied.

So I hope you understand that it is with a considerable amount of dread that I try to make the time to debug this. Please do not hold your breath for it.

**Comment by mintty** [comment]
Johannes, I have no idea why you like to bash on other open source projects and spread the untruth about their issue handling.
As you can follow on https://github.com/mintty/mintty/pull/471, I answered your initial request one day later (during holiday time), and the issue was resolved 10 weeks later. If you call that "months", well... I call that fast, measured by the obscurity and complexity of the involved Windows issue, and compared with serious issues of even more popular open source projects (esp Mozilla) that often stay ignored for years.

I appreciate you concerns to "not blindly update" (https://github.com/git-for-windows/git/issues/580#issuecomment-234764083) being diligent about the stability of Git; but I do not understand why you didn't and don't accept my likewise concerns about the stability and integrity of mintty when I didn't "blindly accept" your pull request but insisted to analyse it, its consequences, and find a proper way to integrate the functionality.

**Comment by dscho** [comment]
I did not do such a thing as spread untruth. Shame on you for suggesting that.

Yes, I call 2.5 months what they are: months. The issue was resolved as far as mintty was concerned. As far as Git for Windows is concerned, the issue is very much open: the fact is that Git for Windows does not work as intended when switching to a non-forked version.

You can appreciate, or not appreciate my concerns, that is your business.

With that, I bow out of this discussion. @khanhoct I am truly sorry that it turns sour this way. Hopefully you can find the time to fight the dragons and figure out what is going wrong, and hopefully you have a truly enjoyable time slaying the monsters. Thanks in advance for all your efforts!

**Comment by mintty** [comment]
> I did not do such a thing as spread untruth. Shame on you for suggesting that.

Shame on you for ignoring the evidence I presented. I hadn't meant to go into more details, in order not to heat the battle you're steering, but you're asking for it:

> We have a track record of problems with this component, 
"track record"? What should that suggest? You were having this taskbar issue which was resolved after a while, and there were some completely unrelated other issues with mintty around the time, e.g. a crash condition that could not positively be attributed to mintty, and some multi-monitor issues that don't affect Git specifically either.

> my PR with the fix was first ignored for a couple of months, 

Come on, this is simply a lie. As I had pointed out before, I responded immediately and then we discussed the issue for some time.

> then it took ages to convince ... I call 2.5 months what they are: months.

Again, I had commented this before. Analysing a weird and complex Window feature to integrate it *properly* rather than in a rash, may take some time. Your expectation that a volunteer maintainer of free software is sitting around just waiting for a nerd who asks for some fancy feature should then react prompty and "blindly" accept your patch is quite an arrogant attitude in the open source scene. Are you a paying customer of mintty? Do you have a 7/24 service contract? I *really* and honestly do not understand your reaction.

**Comment by dscho** [comment]
@khanhoct What is your plan with this issue? Are you interested to work on it more, and have the time?

**Comment by khanhoct** [comment]
@dscho Now, I don't have any solution to overcome this issue.
I have time, it is surely because I'm student *smile*. But I am beginner in programming (I have knowledge of C/C++, Java, OOP, basic skill in using linux, but these skills at basic level) and my English is not good. So I don't know how to help.

I really want to contribute and I'm very pleased if you give me few pointers so I can contribute.

**Comment by dscho** [comment]
@khanhoct thank you! Maybe we could head over to https://gitter.im/git-for-windows/git and discuss things interactively?

**Comment by BladeMight** [comment]
I think icon is missing because _/__root__/bin/bash.exe_ have no icon, if run from _/__root__/git-bash.exe_ icon is, because _/__root__/git-bash.exe_ have it.
P.S. I using portable GIT.

**Comment by dscho** [comment]
@BladeMight good point. @khanhoct how do you start the Git Bash? Via the Start Menu?

**Comment by khanhoct** [comment]
@dscho 
@BladeMight
I tried to run git bash form both paths ( /__root_/bin/bash.exe and /__root_/git-bash.exe). But this issue still exists. Regardless of the installed or portable.

**Comment by BladeMight** [comment]
Try restarting explorer.

**Comment by khanhoct** [comment]
@BladeMight : Git-bash stills is missing icon. **Only git-cmd have icon**

**Comment by BladeMight** [comment]
Are you executing directly from directory by double click on them?

**Comment by dscho** [comment]
This should be fixed with v2.10.0.
